pkg/controller/apis/config/fuzzer/fuzzer.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuzzer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/gofuzz"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	kubectrlmgrconfig "k8s.io/kubernetes/pkg/controller/apis/config"
+)
+
+// Funcs returns the fuzzer functions for the kube-controller manager apis.
+func Funcs(codecs runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		func(obj *kubectrlmgrconfig.KubeControllerManagerConfiguration, c fuzz.Continue) {
+			c.FuzzNoCustom(obj)
+			// add GenericControllerManagerConfiguration values
+			obj.Generic.Address = fmt.Sprintf("%d.%d.%d.%d", c.Intn(256), c.Intn(256), c.Intn(256), c.Intn(256))
+			obj.Generic.MinResyncPeriod = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Hour}
+			obj.Generic.ClientConnection.ContentType = "application/vnd.kubernetes.protobuf"
+			obj.Generic.ClientConnection.QPS = c.Float32()
+			obj.Generic.ClientConnection.Burst = c.Int31()
+			obj.Generic.ControllerStartInterval = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.Generic.LeaderElection.LeaseDuration = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.Generic.LeaderElection.RenewDeadline = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.Generic.LeaderElection.RetryPeriod = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.Generic.LeaderElection.ResourceLock = "endpoints"
+			obj.Generic.LeaderElection.LeaderElect = c.RandBool()
+			obj.Generic.Controllers = []string{"*"}
+			// add KubeCloudSharedConfiguration values
+			obj.KubeCloudShared.NodeMonitorPeriod = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.KubeCloudShared.ClusterName = "kubernetes"
+			obj.KubeCloudShared.ConfigureCloudRoutes = c.RandBool()
+			obj.KubeCloudShared.RouteReconciliationPeriod = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			// add ServiceControllerConfiguration values
+			obj.ServiceController.ConcurrentServiceSyncs = c.Int31()
+			// add AttachDetachControllerConfiguration values
+			obj.AttachDetachController.ReconcilerSyncLoopPeriod = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			// add CSRSigningControllerConfiguration values
+			obj.CSRSigningController.ClusterSigningCertFile = "/etc/kubernetes/ca/ca.pem"
```

Comment:
why is this is not a random string?

Reply:
If set random string, will failed. 
```
roundtrip.go:228: 	round tripping to kubecontrollermanager.config.k8s.io/v1alpha1, Kind=KubeControllerManagerConfiguration v1alpha1.KubeControllerManagerConfiguration
	roundtrip.go:338: KubeControllerManagerConfiguration: diff: 
		object.PersistentVolumeBinderController.VolumeConfiguration.FlexVolumePluginDir:
		  a: ""
		  b: "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/"
```

Reply:
non-empty random strings should roundtrip.

Reply:
using `/+random string` now